test/e2e/12_commit_signing.bats

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env bats
+
+load lib/defer
+load lib/env
+load lib/gpg
+load lib/install
+load lib/poll
+
+function setup() {
+  setup_env
+  kubectl create namespace "${FLUX_NAMESPACE}"
+
+  generate_ssh_secret
+  install_git_srv
+
+  gnupghome=$(tmp_gnupghome)
+  defer rm -rfv "$gnupghome"
+
+  gpg_key=$(create_gpg_key)
+  create_secret_from_gpg_key "$gpg_key"
+  install_flux_gpg "$gpg_key"
+}
+
+@test "Git sync tag is signed" {
+  # Test that a resource from https://github.com/fluxcd/flux-get-started is deployed
+  # This means the Flux instance _should_ have pushed a signed high-watermark tag
+  poll_until_true 'namespace demo' 'kubectl describe ns/demo'
+
+  # Test that the tag has been signed, this errors if this isn't the case
+  pod=$(kubectl --namespace "${FLUX_NAMESPACE}" get pods --no-headers -l app=flux -o custom-columns=":metadata.name" | tail -n 1)
+  kubectl --namespace "${FLUX_NAMESPACE}" exec -it "$pod" \
+    -- sh -c "cd /tmp/flux-gitclone* && git verify-tag flux-sync" >&3
+}
+
+@test "Git commits are signed" {
+  # Assure the resource we are going to lock is deployed
```

Comment:
Ensure that? (not totally sure), I invoke the native-speaking powers of @squaremo

Reply:
They are pretty close in meaning, but "ensure" would be idiomatic here. Typical usage: "I will ensure the door is closed"; "I assure you, the door is closed."